src/moon.c

```diff
@@ -0,0 +1,78 @@
+/*
+
+ The MIT License (MIT)
+
+ Copyright (c) 2023 qlpqlp (twitter.com/inevitable360)
+ Copyright (c) 2022 The Dogecoin Foundation
+
+ Permission is hereby granted, free of charge, to any person obtaining
+ a copy of this software and associated documentation files (the "Software"),
+ to deal in the Software without restriction, including without limitation
+ the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ and/or sell copies of the Software, and to permit persons to whom the
+ Software is furnished to do so, subject to the following conditions:
+ The above copyright notice and this permission notice shall be included
+ in all copies or substantial portions of the Software.
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+ OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ OTHER DEALINGS IN THE SOFTWARE.
+
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+/**
+ * @brief Calling the moon, it returns the current moon fase
+ * @return moon icon
+ */
+
+void *moon()
+{
+  int D,c,e;
+  double jd;                      // Julian days
+
+  time_t t = time(NULL);          // get time
+  struct tm tm = *localtime(&t);  // get localtime
+  int y = tm.tm_year + 1900;      // year, because its only above 1900, we add 1900 to the year
+  int m = tm.tm_mon;              // month
+  int d = tm.tm_mday - 2;         // day, because it can show the moon with 2 days in advance, we subtract
+
+  if (m < 3) {
+    y--;
+    m += 12;
+  }
+  ++m;
+
+  c = 365.25*y;                   // 365.25 total days in a year
+  e = 30.6*m;
+  jd = c+e+d-694039.09;           // Julian days
+  jd /= 29.53;                    // moons cycle 29.53 days
+  D = jd;	                        // convert decimal to int 	   
+  jd -= D;                        // get negative of Julian days
+  D = jd*8 + 0.5;	                // get fraction from 0-8 by rounding 0.5 to it               
+
+  switch (D & 7) {                // because the new moon = 0 is the same on 8 we convert to 0
+    case 0:
+      return "🌚";               // New Moon Hype / Discounts
+    case 1:
+      return "🐶";               // Show Doge wen no moon
+    case 2:
+      return "🌓";               // First Quarter Moon
+    case 3:
+      return "🐶";               // Show Doge wen no moon
+    case 4:
+      return "🌝";               // Full Moon DownTrend / No Discounts
+    case 5:
+      return "🐶";               // Show Doge wen no moon
+    case 6:
+      return "🌗";               // Third Quarter
+    case 7:
+      return "🐶";               // Show Doge wen no moon
+  }
+}
```

Comment:
New line

Reply:
> This looks super cool, would you be willing to add to `Makefile.am`, `CMakeLists.txt` and perhaps write a test to assert your results?
> 
> ```
> touch test/moon_tests.c
> ```
> 
> and add to `test/unittester.c`?
> 
Never done it, but will try to do eheh :P

Reply:
Let me know if you have questions

Reply:
> Let me know if you have questions

lol, need some guidance, sorry :P

Reply:
No prob. I'm going to check it out shortly, this looks p good though!

Reply:
Alright so I struggled to come up with something re your tests or what exactly we should be testing there so as long as it's (`char* moon`) not returning false or 0 I think we should be fine? If it's based on current timing I don't think testing the previous way is the right path to go so I'd like to work with you on what we could come up with.

https://github.com/xanimo/libdogecoin/commits/e980e02bb5499c81413ef7ccc83adc98be5cfa24

Outside of that, I'd like to get the rest of the commits cleaned up so do you know how to interactive rebase?

Reply:
Dono why is failing compiling lol 
Wen u have time to see why I appreciate, I'm a litle stuck :)
![image](https://user-images.githubusercontent.com/5968076/218268743-cc8affc5-1974-48b0-84be-49e7158cc203.png)

Reply:
Hey, so if you can run the following and force push should be fine:
```
git checkout 0.1.2-dev
git branch -m 0.1.2-dev-moon-backup
git fetch upstream
git checkout -b 0.1.2-dev upstream/0.1.2-dev
```

Then just copy the files from: https://github.com/xanimo/libdogecoin/commit/8d262ac27327994a552220ce299c07f7f041ad16

And then:
```
git push -f origin 0.1.2-dev
```
And we should be good <3

Reply:
So ardently the problem was on my test lol

Much thanks for taking your time to look at everything, teach me an even code for this to be in Libdogecoin :)

Reply:
Alright my friend, apologies we have to do this one more time:
```
git checkout 0.1.2-dev
git fetch upstream
git reset --hard HEAD~1
```
Add the following changes (copy/paste):
https://github.com/xanimo/libdogecoin/commit/7ac420b57c2ea0160fcf89aaeca66f149226d645

Then:
```
git add .
git commit
git push -f origin 0.1.2-dev
```